ddl/backfilling.go

```diff
@@ -357,59 +373,65 @@ func splitTableRanges(t table.PhysicalTable, store kv.Storage, startKey, endKey
 	return ranges, nil
 }
 
-func waitTaskResults(workers []*backfillWorker, taskCnt int,
-	totalAddedCount *int64, startKey kv.Key) (kv.Key, int64, error) {
+const adjustWorkerSizeInterval = 16
+
+func waitTaskResults(scheduler *backfillScheduler, batchTasks []*reorgBackfillTask,
+	totalAddedCount *int64) (kv.Key, int64, error) {
 	var (
 		addedCount int64
-		nextKey    = startKey
 		firstErr   error
 	)
-	for i := 0; i < taskCnt; i++ {
-		worker := workers[i]
-		result := <-worker.resultCh
+	keeper := newDoneTaskKeeper(batchTasks[0].startKey)
+	taskSize := len(batchTasks)
+	for i := 0; i < taskSize; i++ {
+		result := <-scheduler.resultCh
 		if firstErr == nil && result.err != nil {
 			firstErr = result.err
 			// We should wait all working workers exits, any way.
 			continue
 		}
 
 		if result.err != nil {
-			logutil.BgLogger().Warn("[ddl] backfill worker failed", zap.Int("workerID", worker.id),
+			logutil.BgLogger().Warn("[ddl] backfill worker failed",
+				zap.String("result next key", hex.EncodeToString(result.nextKey)),
 				zap.Error(result.err))
 		}
 
 		if firstErr == nil {
 			*totalAddedCount += int64(result.addedCount)
 			addedCount += int64(result.addedCount)
-			nextKey = result.nextKey
+			keeper.updateNextKey(result.taskID, result.nextKey)
+			if i%adjustWorkerSizeInterval == 0 {
```

Comment:
Not get point here. What is the purpose for `i%adjustWorkerSizeInterval == 0`?

Reply:
It is used to prevent querying global_variables too frequently. Previously the interval is `4`.

Reply:
Could some comments here?